### PR TITLE
docs: improve monitoring documentation

### LINKED ISSUE #62: Recommendations for alerting on failed releases
I'd like to be able to alert on failed releases. I looked into hooking into the kubernetes events that the operator emits, but the only event emitted is `chartsynced` and it doesn't contain any error information.

```
Object:  { kind: 'Event',
  apiVersion: 'v1',
  metadata:
   { name: 'bad-hr-test.15cada07d1ccac62',
     namespace: 'new-ns',
     selfLink:
      '/api/v1/namespaces/new-ns/events/bad-hr-test.15cada07d1ccac62',
     uid: '507be1a7-e7ae-11e9-9058-42010a8000c0',
     resourceVersion: '1006825',
     creationTimestamp: '2019-10-05T20:25:47Z' },
  involvedObject:
   { kind: 'HelmRelease',
     namespace: 'new-ns',
     name: 'bad-hr-test',
     uid: '5077e09e-e7ae-11e9-9058-42010a8000c0',
     apiVersion: 'flux.weave.works/v1beta1',
     resourceVersion: '39704805' },
  reason: 'ChartSynced',
  message: 'Chart managed by HelmRelease processed',
  source: { component: 'helm-operator' },
  firstTimestamp: '2019-10-05T20:25:47Z',
  lastTimestamp: '2019-10-05T20:35:31Z',
  count: 7,
  type: 'Normal',
  eventTime: null,
  reportingComponent: '',
  reportingInstance: '' }
```

The only way I've found to discover bad releases is by introspecting the logs of the helm-operator. For example:
```
ts=2019-10-05T20:30:13.497815163Z caller=release.go:208 component=release error="Chart release failed: bad-hr-test: &status.statusError{Code:2, Message:\"render error in \\\"blah/templates/ui/deployment.yaml\\\": template: blah/templates/ui/deployment.yaml:21:55: executing \\\"blah/templates/ui/deployment.yaml\\\" at <required \\\"ui image tag required\\\" .Values.ui.image.tag>: error calling required: ui image tag required\", Details:[]*any.Any(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}"
```

This sometimes appears the Status of the helm release, but it looks like sometimes isn't reported:

```
Name:         bad-hr-test
Namespace:    new-ns
Labels:       <none>
Annotations:  flux.weave.works/automated: false
              kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"flux.weave.works/v1beta1","kind":"HelmRelease","metadata":{"annotations":{"flux.weave.works/automated":"false"},"name":"bad...
API Version:  flux.weave.works/v1beta1
Kind:         HelmRelease
Metadata:
  Creation Timestamp:  2019-10-05T20:25:47Z
  Generation:          1
  Resource Version:    39705744
  Self Link:           /apis/flux.weave.works/v1beta1/namespaces/new-ns/helmreleases/bad-hr-test
  UID:                 5077e09e-e7ae-11e9-9058-42010a8000c0
Spec:
  Chart:
    Git:         git@github.com:something/cluster-config
    Path:        charts/blah
  Release Name:  bad-hr-test
Status:
  Conditions:
    Last Transition Time:  2019-10-05T20:30:08Z
    Message:               successfully cloned git repo
    Reason:                GitRepoCloned
    Status:                True
    Type:                  ChartFetched
    Last Transition Time:  2019-10-05T20:30:13Z
    Message:               rpc error: code = Unknown desc = release: "bad-hr-test" not found
    Reason:                HelmInstallFailed
    Status:                False
    Type:                  Released
```

I can think of a couple of different ways this might get solved:
- Publishing more kubernetes events 
- adding a `failed install` status to `release status` (currently it will only report a status if the install actually happened, not if there's a render issue)

If you're in favour of adding more kube events, how best should I add the eventBroader to the release.go and other helm related files?

Thanks!

-Kyle

### LINKED ISSUE #381: Clarify meaning of metrics on monitoring.md
Hi,

I'm reading https://github.com/fluxcd/helm-operator/blob/master/docs/references/monitoring.md and the difference between release_phase_duration_seconds and release_duration_seconds isn't very clear to me.  The only difference seems to be "Release" and "Release Phase", which I'm not sure where they are defined.  Is it possible to add some clarification comments around how the two metrics are different?

Thanks!

### PR DESCRIPTION
**Pull request by hiddeco**
Fixes #381.
Fixes #62.
Documents #383.

### CODE REVIEW

**Comment by stefanprodan** [comment]
I think we should include the list:
```
	// Unknown is mapped to 0
	v1.HelmReleasePhaseChartFetchFailed: -4,
	v1.HelmReleasePhaseFailed:           -3,
	v1.HelmReleasePhaseRollbackFailed:   -2,
	v1.HelmReleasePhaseRolledBack:       -1,
	v1.HelmReleasePhaseRollingBack:      1,
	v1.HelmReleasePhaseInstalling:       2,
	v1.HelmReleasePhaseUpgrading:        3,
	v1.HelmReleasePhaseChartFetched:     4,
	v1.HelmReleasePhaseSucceeded:        5,

```

**Comment by stefanprodan** [comment]
We could also include an example of how to alert on failed releases, similar to flux docs: https://docs.fluxcd.io/en/1.19.0/references/monitoring/

**Comment by hiddeco** [comment]
@stefanprodan yes, that was also my initial though but I was having a hard time finding the right formatting for all those values. Given the document is simple and table based at the moment.

**Comment by onedr0p** [comment]
@hiddeco You can include _See phases below_ link and list them outside the table

**Comment by sa-spag** [comment]
If you want to mention Prometheus alert rules, I suggest the following:

- the Helm Operator struggles to process its release queue in a timely manner:
```
alert: HelmOperatorLowThroughput
expr: flux_helm_operator_release_queue_length_count > 0
for: 30m
```

- a HelmRelease was automatically rolled back:
```
alert: HelmReleaseRolledBack
expr: flux_helm_operator_release_phase_info == -1
```

- a HelmRelease is subject to an error:
```
alert: HelmReleaseError
expr: flux_helm_operator_release_phase_info < -1
```

Note that these does not cover Helm releases statuses as exposed by [sstarcher/helm-exporter](https://github.com/sstarcher/helm-exporter) (you might want to include that notice in the docs).

File: docs/references/monitoring.md [hunk 1]
```diff
@@ -4,10 +4,58 @@ The Helm Operator exposes a metrics endpoint at `/metrics`  on the configured
...
+```yaml
+alert: HelmOperatorLowThroughput
+expr: flux_helm_operator_release_queue_length_count > 0
+for: 30m
+```
+
+### Automatic rollback of `HelmRelease`
+
+```yaml
+alert: HelmReleaseError
```
Comment (sa-spag) [inline docs/references/monitoring.md]:
```suggestion
alert: HelmReleaseRolledBack
```

**Review by stefanprodan — APPROVED** [review]
LGTM 💯

**Pull request merged**